shap/actions/_optimizer.py

```diff
@@ -1,23 +1,24 @@
 import copy
 import queue
 import warnings
+from typing import Union
 
 from ..utils._exceptions import ConvergenceError, InvalidAction
 from ._action import Action
 
 
 class ActionOptimizer:
-    def __init__(self, model, actions):
+    def __init__(self, model, actions: list[Union[Action, list[Action]]]):
         self.model = model
-        warnings.warn("Note that ActionOptimizer is still in an alpha state and is subjust to API changes.")
+        warnings.warn("Note that ActionOptimizer is still in an alpha state and is subject to API changes.")
         # actions go into mutually exclusive groups
-        self.action_groups = []
+        self.action_groups: list[list[Action]] = []
         for group in actions:
-            if issubclass(type(group), Action):
+            if isinstance(group, Action):
```

Comment:
I believe `isinstance(foo, Bar)` is equivalent to `issubclass(type(foo), Bar)`, but it would be useful to have a second opinion on this.

Reply:
Hmm, the only thing that I could think of is to not match instances of `Action` but just subclasses of it but `subclass` does not behave like this

```python
>>> class A:
...     pass
... 
>>> class B(A):
...     pass
... 
>>> a = A()
>>> b = B()
>>> issubclass(type(b), A)
True
>>> issubclass(type(a), A) 
True  # I expected this to be False
```

So I guess the only reason here was that someone didn't know better.